osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -46,36 +46,41 @@ public class MasterGameplayClockContainer : GameplayClockContainer
 
         private double totalAppliedOffset => userBeatmapOffsetClock.RateAdjustedOffset + userGlobalOffsetClock.RateAdjustedOffset + platformOffsetClock.RateAdjustedOffset;
 
-        private readonly BindableDouble pauseFreqAdjust = new BindableDouble(1);
+        private readonly BindableDouble pauseFreqAdjust = new BindableDouble(); // Important that this starts at zero, matching the paused state of the clock.
 
         private readonly WorkingBeatmap beatmap;
-        private readonly double gameplayStartTime;
-        private readonly bool startAtGameplayStart;
-        private readonly double firstHitObjectTime;
 
         private HardwareCorrectionOffsetClock userGlobalOffsetClock;
         private HardwareCorrectionOffsetClock userBeatmapOffsetClock;
         private HardwareCorrectionOffsetClock platformOffsetClock;
         private MasterGameplayClock masterGameplayClock;
         private Bindable<double> userAudioOffset;
-        private double startOffset;
 
         private IDisposable beatmapOffsetSubscription;
 
+        private readonly double latestGameplayStartTime;
+
         [Resolved]
         private RealmAccess realm { get; set; }
 
         [Resolved]
         private OsuConfigManager config { get; set; }
 
-        public MasterGameplayClockContainer(WorkingBeatmap beatmap, double gameplayStartTime, bool startAtGameplayStart = false)
+        /// <summary>
+        /// Create a new master gameplay clock container.
+        /// </summary>
+        /// <param name="beatmap">The beatmap to be used for time and metadata references.</param>
+        /// <param name="latestGameplayStartTime">The latest time which should be used when introducing gameplay. Will be used when skipping forward.</param>
+        /// <param name="startFromLatestStartTime">Whether to start from the provided latest start time rather than zero.</param>
```

Comment:
I don't follow what this `startFromLatestStartTime` toggle is for. When would you want/not want this behaviour? I tried to infer from usages but it looks almost entirely random.

Reply:
Only thing I can gather is it looks like making up for `gameplayStartTime` not being a `double?`... Definitely weird and xmldoc doesn't help.

Also not sure why there are 3 ways of providing the start time - via `MasterGameplayClock` ctor, via `StartTime`, and via `Reset()`.

Reply:
I'll look into the final parameter as I agree it's a bit weird and usages are hard to follow.

> Also not sure why there are 3 ways of providing the start time - via `MasterGameplayClock` ctor, via `StartTime`, and via `Reset()`.

The ctor is setting `latestGameplayStartTime`, which has a very different meaning and purpose. Can you double-check the xmldoc and confirm you can understand from it? The ctor one doesn't change post-construction and is used for cases like skipping forward to gameplay.

Reply:
How about `skipTargetTime`?

Reply:
> I don't follow what this `startFromLatestStartTime` toggle is for. When would you want/not want this behaviour? I tried to infer from usages but it looks almost entirely random.

I've removed this and updated usages. Hopefully the local usages should be more understandable due to `StartTime` being set directly.

Reply:
New naming and xmldocs are much better.